src/FuncTorrent/Bencode.hs

```diff
@@ -145,8 +145,10 @@ bencVal = Bstr <$> bencStr <|>
           Blist <$> bencList <|>
           Bdict <$> bencDict
 
-decode :: ByteString -> Either ParseError BVal
-decode = parse bencVal "BVal"
+decode :: ByteString -> Either String BVal
+decode bs = case (parse bencVal "BVal" bs) of
+           Left e -> Left "Unable to parse torrent file"
```

Comment:
Parse error is useless to the end user. What we need is debug logs + message for user. Purity is a pain in this case.